test/abstract_syntax_tree.coffee

```diff
@@ -219,40 +607,101 @@ test "AST as expected for ModuleSpecifierList node", ->
   # TODO
 
 test "AST as expected for ImportSpecifierList node", ->
-  # TODO
+  testExpression 'import React, {Component} from "react"',
+    clause: namedImports:
+      type: 'ImportSpecifierList'
+      specifiers: [
+        identifier: 'Component'
+      ]
 
 test "AST as expected for ExportSpecifierList node", ->
-  # TODO
+  testExpression 'export {a, b, c}',
+    clause:
+      type: 'ExportSpecifierList'
+      specifiers: [
+        {identifier: 'a'}
+        {identifier: 'b'}
+        {identifier: 'c'}
+      ]
 
 test "AST as expected for ModuleSpecifier node", ->
   # TODO
 
 test "AST as expected for ImportSpecifier node", ->
-  # TODO
+  testExpression 'import {Component, PureComponent} from "react"',
+    clause: namedImports: specifiers: [
+      type: 'ImportSpecifier'
+      identifier: 'Component'
+      moduleDeclarationType: 'import'
+      original:
+        type: 'IdentifierLiteral'
+        value: 'Component'
+    ,
+      type: 'ImportSpecifier'
+      identifier: 'PureComponent'
+    ]
 
 test "AST as expected for ImportDefaultSpecifier node", ->
-  # TODO
+  testExpression 'import React from "react"',
+    clause: defaultBinding:
+      type: 'ImportDefaultSpecifier'
+      moduleDeclarationType: 'import'
+      identifier: 'React'
+      original: type: 'IdentifierLiteral'
 
 test "AST as expected for ImportNamespaceSpecifier node", ->
   # TODO
 
 test "AST as expected for ExportSpecifier node", ->
-  # TODO
+  testExpression 'export {X}',
+    clause: specifiers: [
+      type: 'ExportSpecifier'
+      moduleDeclarationType: 'export'
+      identifier: 'X'
+      original: type: 'IdentifierLiteral'
+    ]
 
 test "AST as expected for Assign node", ->
-  # TODO
+  testExpression 'a = 1',
+    type: 'Assign'
+    variable: value: 'a'
+    value: value: '1'
+
+  testExpression 'a: 1',
+    properties: [
+      type: 'Assign'
+      context: 'object'
+      originalContext: 'object'
+      variable: value: 'a'
+      value: value: '1'
+    ]
 
 test "AST as expected for FuncGlyph node", ->
   # TODO
 
 test "AST as expected for Code node", ->
-  # TODO
+  testExpression '=>',
+    type: 'Code'
+    bound: yes
+    body: type: 'Block'
 
 test "AST as expected for Param node", ->
-  # TODO
+  testExpression '(a = 1) ->',
+    params: [
+      type: 'Param'
+      name: value: 'a'
+      value: value: '1'
+    ]
 
 test "AST as expected for Splat node", ->
-  # TODO
+  testExpression '(a...) ->',
+    params: [
+      type: 'Param'
+      splat: yes
+      name: value: 'a'
+    ]
+
+  # TODO: Test object splats.
 
 test "AST as expected for Expansion node", ->
   # TODO
```

Comment:
Can we add this one, and the following?

Reply:
`Expansion` is defined in a comment as “Used to skip values inside an array destructuring (pattern matching) or parameter list.” I wonder if this node is even used anymore, now that we use ES destructuring.

Reply:
`Elision` is just `[,a, , , b, , c, ,]` (the skipped elements). Is that not output in the AST?